source/jobs/JobsFeature.cpp

```diff
@@ -590,11 +593,22 @@ void JobsFeature::executeJob(JobExecutionData job)
             shutdownHandler);
         return;
     }
-    else
-    {
-        publishUpdateJobExecutionStatus(job, {Iotjobs::JobStatus::IN_PROGRESS, "", "", ""});
-    }
+    publishUpdateJobExecutionStatus(job, {Iotjobs::JobStatus::IN_PROGRESS, "", "", ""});
```

Comment:
IMO: Nicer to create a single argument constructor for `JobExecutionStatusInfo`.

Reply:
So, there are several different instantiations of JobExecutionStatusInfo. I can create a single argument constructor for this case but was wondering if there is any reason not to make it single argument with the remaining arguments default "" so that it can be constructed in all the different ways with same constructor (status), (status, "reason"), (status, "", "stdout"), etc.

Reply:
By default std::string will construct empty string. You can create 1-argument (status-only), 2-argument (status,  string), 3-argument (status, string, string) constructors and leave the other strings unset since they will be default constructed to empty string.

Reply:
So, when I went to do this I realized there is never both a stderror and stdout so I made a single argument constructor and then an all-args constructor. Refactored a bit so that when publishing statusDetails it will place the output correctly.